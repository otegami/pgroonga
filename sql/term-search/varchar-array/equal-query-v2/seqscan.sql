CREATE TABLE tags (
  names varchar[]
);

INSERT INTO tags VALUES (ARRAY['PostgreSQL', 'PG']);
INSERT INTO tags VALUES (ARRAY['Groonga', 'grn', 'groonga']);
INSERT INTO tags VALUES (ARRAY['PGroonga', 'pgrn', 'SQL']);

SET enable_seqscan = on;
SET enable_indexscan = off;
SET enable_bitmapscan = off;

EXPLAIN (COSTS OFF)
SELECT names
  FROM tags
 WHERE names &=~ 'grn OR sql';

SELECT names
  FROM tags
 WHERE names &=~ 'grn OR sql';

DROP TABLE tags;
