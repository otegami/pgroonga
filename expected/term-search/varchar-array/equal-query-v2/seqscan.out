CREATE TABLE tags (
  names varchar[]
);
INSERT INTO tags VALUES (ARRAY['PostgreSQL', 'PG']);
INSERT INTO tags VALUES (ARRAY['Groonga', 'grn', 'groonga']);
INSERT INTO tags VALUES (ARRAY['PGroonga', 'pgrn', 'SQL']);
SET enable_seqscan = on;
SET enable_indexscan = off;
SET enable_bitmapscan = off;
EXPLAIN (COSTS OFF)
SELECT names
  FROM tags
 WHERE names &=~ 'grn OR sql';
                QUERY PLAN                
------------------------------------------
 Seq Scan on tags
   Filter: (names &=~ 'grn OR sql'::text)
(2 rows)

SELECT names
  FROM tags
 WHERE names &=~ 'grn OR sql';
         names         
-----------------------
 {Groonga,grn,groonga}
 {PGroonga,pgrn,SQL}
(2 rows)

DROP TABLE tags;
